inbm/dispatcher-agent/dispatcher/sota/os_updater.py

```diff
@@ -168,6 +177,28 @@ def _get_estimated_size_from_apt_get_upgrade(upgrade_output: str) -> int:
             return 0
 
 
+    def no_download(self):
```

Comment:
Check the existing method that does the full update to see how we do it in the 'full' mode. Also consider whether you want to reboot after no-download/download-only and how these modes interact with logging the update. Finally, if download-only is run and you're not rebooting, make sure we aren't going to health check/roll back if there's a reboot (since we only ran a download), at least not until we apply the update later in no-download mode.

Reply:
@gblewis1 the reboot part is handled in other update by yeng Liong. we are not looking at it in this story

Reply:
> @gblewis1 the reboot part is handled in other update by yeng Liong. we are not looking at it in this story

That's not how I read the story. https://rtc.intel.com/ccm0006001/web/projects/SSEA-%20IOTG%20PED%20Team#action=com.ibm.team.workitem.viewWorkItem&id=530032 has a comment saynig  'Do ensure that if the sita cmd is a "download-only", the device should NOT go through a reboot' and there is a child story with title '[LP] Do ensure that if the sota cmd is a "download-only", the device should NOT go through a reboot'.  @pillaiabhishek has this changed?

Reply:
@gblewis1 yes it is @pillaiabhishek who got that part assigned to Yeng Liong, he is currently working on that MR. This MR does not handle reboot functionality.

Reply:
I dont see a reason why a device needs to go through a reboot if in a "download-only" sota cmd.
Any concerns here @tsirlapu @gblewis1 ?

Reply:
@tsirlapu, Yeng Liong needs to work on adding a --reboot option in the SOTA cmd.
But even without that, in this PR we need to ensure that if a user sends a SOTA "download-only" cmd, the device does not reboot.

Below is expected from the PR:
1. inbc sota _-> Perform download and install. Reboots the platform._
2. inbc sota --command update _-> Perform download and install. Reboots the platform._
3. inbc sota --command update --mode download-only _-> Perform download only. No platform reboot._
4. inbc sota --command update --mode no-download _-> Perform install only. Reboots the platform._

Reply:
@pillaiabhishek this is already discussed, This MR is without the reboot functionality, the next MR is with Reboot functionality,  here are the changes for reboot https://github.com/intel/intel-inb-manageability/pull/345

Reply:
@gblewis1 we are good, this MR does not cover reboot functionality changes, @pillaiabhishek just confirmed as discussed in our email chain "sounds good Teja. 
Do let Gavin know that Yeng Liong already has that covered." Changes of reboot covered in MR345